stdlib/regex.gr

```diff
@@ -0,0 +1,2983 @@
+}
+
+let lineStartMatcher = (next_m) => (buf: MatchBuf, pos: Number, start: Number, limit: Number, end: Number, state, stack) => {
+  if (pos == start || matchBufChar(buf, pos - 1) == Ok('\n')) next_m(buf, pos, start, limit, end, state, stack) else None
+}
+
+let lineEndMatcher = (next_m) => (buf: MatchBuf, pos: Number, start: Number, limit: Number, end: Number, state, stack) => {
+  if (pos == end || matchBufChar(buf, pos) == Ok('\n')) next_m(buf, pos, start, limit, end, state, stack) else None
+}
+
+let isWordChar = (c) => {
+  match(c) {
+    Err(_) => false,
+    Ok(c) when (Char.code('0') <= Char.code(c) && Char.code(c) <= Char.code('9')) => true,
+    Ok(c) when (Char.code('a') <= Char.code(c) && Char.code(c) <= Char.code('z')) => true,
+    Ok(c) when (Char.code('A') <= Char.code(c) && Char.code(c) <= Char.code('Z')) => true,
+    Ok(c) when (Char.code('_') <= Char.code(c)) => true,
+    _ => false
+  }
+}
+
+let isWordBoundary = (buf, pos, start, limit, end) => {
+  !((pos == start || !isWordChar(matchBufChar(buf, pos - 1))) == (pos == end || !isWordChar(matchBufChar(buf, pos))))
+}
+
+let wordBoundaryMatcher = (next_m) => (buf: MatchBuf, pos: Number, start: Number, limit: Number, end: Number, state, stack) => {
+  if (isWordBoundary(buf, pos, start, limit, end)) next_m(buf, pos, start, limit, end, state, stack) else None
+}
+
+let notWordBoundaryMatcher = (next_m) => (buf: MatchBuf, pos: Number, start: Number, limit: Number, end: Number, state, stack) => {
+  if (!isWordBoundary(buf, pos, start, limit, end)) next_m(buf, pos, start, limit, end, state, stack) else None
+}
+
+// Alternatives
+
+let altsMatcher = (m1, m2) => (buf: MatchBuf, pos: Number, start: Number, limit: Number, end: Number, state, stack) => {
+  match(m1(buf, pos, start, limit, end, state, stack)) {
+    None => m2(buf, pos, start, limit, end, state, stack),
+    Some(v) => Some(v)
+  }
+}
+
+// repeats, greedy (default) and non-greedy
+
+let repeatMatcher = (r_m, min, max, next_m) => (buf: MatchBuf, pos: Number, start: Number, limit: Number, end: Number, state, stack) => {
+  let rec rloop = (pos, n) => {
+    if (n < min) {
+      let newStack = [SEPositionProducer(pos => rloop(pos, n + 1)), ...stack]
+      r_m(buf, pos, start, limit, end, state, newStack)
+    } else if (match(max) { None => false, Some(max) => max == n}) {
+      next_m(buf, pos, start, limit, end, state, stack)
+    } else {
+      let newStack = [SEPositionProducer(pos => rloop(pos, n + 1)), ...stack]
+      match(r_m(buf, pos, start, limit, end, state, newStack)) {
+        Some(v) => Some(v),
+        None => next_m(buf, pos, start, limit, end, state, stack)
+      }
+    }
+  }
+  rloop(pos, 0)
+}
+
+let rStack = [SEPositionProducer(pos => Some(pos))]
+
+let arrayCopy = (dest, destStart, src, srcStart, srcEnd) => {
+  let mut count = srcStart
+  while (count < srcEnd) {
+    dest[destStart + (count - srcStart)] = src[count]
+    count = count + 1
+  }
+}
```

Comment:
I don't see it in there. I didn't add it because I wasn't sure if we wanted this precise API included in `Array`. Let's leave that to a separate PR.

Reply:
Please make an issue 🙏

Reply:
@phated What should the issue be? We need to discuss whether we want to have the API and what the design should be, so it wouldn't be appropriate to have an issue saying to implement it.

Reply:
How about "Implement Array.fillSlice"? That seems to be aligned with our Buffer API naming

Reply:
Again, my point is that the question of "should we do this" is one which we should discuss, rather than directly opening an issue saying "Implement this thing". For example, I don't immediately see how `fillSlice` is in line with buffer's naming. I would say that `move` makes more sense, since `Bytes.move` does essentially the same thing as this function.